packages/dashboard/src/dashboard-actions/actions.ts

```diff
@@ -26,11 +29,196 @@ export interface ResizeAction extends Action<'RESIZE'> {
     changeInPosition: Position;
     widgetIds: string[];
     cellSize: number;
+    dashboardConfiguration: DashboardConfiguration;
+    isActionComplete: boolean;
   };
 }
 export const onResizeAction = (payload: ResizeAction['payload']): ResizeAction => ({
   type: RESIZE,
   payload,
 });
 
-export type DashboardAction = MoveAction | ResizeAction;
+export type ResizeActionInput = ResizeAction['payload'];
+
+export const DELETE = 'DELETE';
+
+export interface DeleteAction extends Action<'DELETE'> {
+  type: typeof DELETE;
+  payload: {
+    widgetIds: string[];
+    widgets: Widget[];
+  };
+}
+export const onDeleteAction = (payload: DeleteAction['payload']): DeleteAction => ({
+  type: DELETE,
+  payload,
+});
+
+export type DeleteActionInput = DeleteAction['payload'];
+
+export const PASTE = 'PASTE';
+
+export interface PasteAction extends Action<'PASTE'> {
+  type: typeof PASTE;
+}
+
+export type PasteActionInput = PasteAction;
+
+export const onPasteAction = (): PasteAction => ({
+  type: PASTE,
+});
+
+export const CREATE = 'CREATE';
+
+export interface CreateAction extends Action<'CREATE'> {
+  type: typeof CREATE;
+  payload: {
+    dashboardConfiguration: DashboardConfiguration;
+    widgets: Widget[];
+  };
+}
+
+export const onCreateAction = (payload: CreateAction['payload']): CreateAction => ({
+  type: CREATE,
+  payload,
+});
+
+export type CreateActionInput = CreateAction['payload'];
+
+export const REDO = 'REDO';
+
+export interface RedoAction extends Action<'REDO'> {
+  type: typeof REDO;
+  payload: {
+    redoAction: DashboardAction;
+  };
+}
+
+export type RedoActionInput = RedoAction['payload'];
+
+export const onRedoAction = (payload: RedoActionInput): RedoAction => ({
+  type: REDO,
+  payload,
+});
+
+export const COPY = 'COPY';
+
+export interface CopyAction extends Action<'COPY'> {
+  type: typeof COPY;
+  payload: {
+    copyGroup: Widget[];
+  };
+}
+
+export type CopyActionInput = CopyAction['payload'];
+
+export const onCopyAction = (payload: CopyActionInput): CopyAction => ({
+  type: COPY,
+  payload,
+});
+
+export const UNDO = 'UNDO';
+
+export interface UndoAction extends Action<'UNDO'> {
+  type: typeof UNDO;
+  payload: {
+    undoAction: DashboardAction;
+  };
+}
+
+export type UndoActionInput = UndoAction['payload'];
+
+export const onUndoAction = (payload: UndoActionInput): UndoAction => ({
+  type: UNDO,
+  payload,
+});
+
+export const REDOPASTE = 'REDOPASTE';
+
+export interface RedoPasteAction extends Action<'REDOPASTE'> {
+  type: typeof REDOPASTE;
+  payload: {
+    numTimesPasted: number;
+    copyGroup: Widget[];
+  };
+}
+
+export type RedoPasteActionInput = RedoPasteAction['payload'];
+
+export const onRedoPasteAction = (payload: RedoPasteActionInput): RedoPasteAction => ({
```

Comment:
what uses this? wouldn't the dashboard just blindly call `undo`, and internally it would delegate?